File: masiwasi/app/src/main/java/models/Mascota.java
package models;

import java.io.Serializable;

public class Mascota implements Serializable {

    private String id;
    private String nombre;
    private String edad;
    private String sexo;
    private String descripcion;
    private String categoria;
    private String color;
    private String imageUrl;

    public Mascota() {
        // Constructor vacío requerido para Firestore
    }

    public Mascota(String id, String nombre, String edad, String sexo,
                   String descripcion, String categoria, String color,
                   String imageUrl) {

        this.id = id;
        this.nombre = nombre;
        this.edad = edad;
        this.sexo = sexo;
        this.descripcion = descripcion;
        this.categoria = categoria;
        this.color = color;
        this.imageUrl = imageUrl;
    }

    // Getters y setters

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getNombre() { return nombre; }
    public void setNombre(String nombre) { this.nombre = nombre; }

    public String getEdad() { return edad; }
    public void setEdad(String edad) { this.edad = edad; }

    public String getSexo() { return sexo; }
    public void setSexo(String sexo) { this.sexo = sexo; }

    public String getDescripcion() { return descripcion; }
    public void setDescripcion(String descripcion) { this.descripcion = descripcion; }

    public String getCategoria() { return categoria; }
    public void setCategoria(String categoria) { this.categoria = categoria; }

    public String getColor() { return color; }
    public void setColor(String color) { this.color = color; }

    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }
}

File: movil2/mozita/app/src/main/java/models/Mascota.kt
package models

import java.io.Serializable

class Mascota : Serializable {

    private var id: String? = null
    private var nombre: String? = null
    private var edad: String? = null
    private var sexo: String? = null
    private var descripcion: String? = null
    private var categoria: String? = null
    private var color: String? = null
    private var imageUrl: String? = null
    private var duenoId: String? = null

    // Constructor vacío requerido para Firestore
    constructor()

    constructor(
        id: String?,
        nombre: String?,
        edad: String?,
        sexo: String?,
        descripcion: String?,
        categoria: String?,
        color: String?,
        imageUrl: String?,
        duenoId: String?
    ) {
        this.id = id
        this.nombre = nombre
        this.edad = edad
        this.sexo = sexo
        this.descripcion = descripcion
        this.categoria = categoria
        this.color = color
        this.imageUrl = imageUrl
        this.duenoId = duenoId
    }

    // Getters y setters

    fun getId(): String? {
        return id
    }

    fun setId(id: String?) {
        this.id = id
    }

    fun getNombre(): String? {
        return nombre
    }

    fun setNombre(nombre: String?) {
        this.nombre = nombre
    }

    fun getEdad(): String? {
        return edad
    }

    fun setEdad(edad: String?) {
        this.edad = edad
    }

    fun getSexo(): String? {
        return sexo
    }

    fun setSexo(sexo: String?) {
        this.sexo = sexo
    }

    fun getDescripcion(): String? {
        return descripcion
    }

    fun setDescripcion(descripcion: String?) {
        this.descripcion = descripcion
    }

    fun getCategoria(): String? {
        return categoria
    }

    fun setCategoria(categoria: String?) {
        this.categoria = categoria
    }

    fun getColor(): String? {
        return color
    }

    fun setColor(color: String?) {
        this.color = color
    }

    fun getImageUrl(): String? {
        return imageUrl
    }

    fun setImageUrl(imageUrl: String?) {
        this.imageUrl = imageUrl
    }

    fun getDuenoId(): String? {
        return duenoId
    }

    fun setDuenoId(duenoId: String?) {
        this.duenoId = duenoId
    }
}